design/workload-labeling.md

```diff
@@ -0,0 +1,236 @@
+kind: ApplicationConfiguration
+metadata:
+  name: appconfig-example
+spec:
+  components:
+    - componentName: php-apache
+      traits:
+        - trait:
+            apiVersion: autoscaling/v2beta2
+            kind: HorizontalPodAutoscaler
+            metadata:
+              name: php-apache
+            spec:
+              scaleTargetRef:
+                apiVersion: apps/v1
+                kind: Deployment
+                name: php-apache
+---
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: php-apache
+spec:
+  workload:
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: php-apache
+```
+
+3). Here is a similar situation in which users have to manually set `serviceName` and `servicePort` in trait Ingress.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: ApplicationConfiguration
+metadata:
+  name: example
+spec:
+  components:
+    - componentName: web
+      traits:
+        - trait:
+            apiVersion: v1
+            kind: Service
+            metadata:
+              name: web
+            spec:
+              selector:
+                app: web
+              ports:
+              - port: 80
+        - trait:
+            apiVersion: extensions/v1beta1
+            kind: Ingress
+            spec:
+              rules:
+              - http:
+                  paths:
+                  - path: /
+                    backend:
+                      serviceName: web
+                      servicePort: 80
+```
+
+4). If a trait can easily retrieve the underlying pods, it will be helpful for:
+- ingress/service/traffic trait can easily route their traffic to pods with the information of OAM AppConfig/Component.
+- log/metrics trait can easily find which pods need to gather logs or metrics by OAM AppConfig.
+At present, users have to manually set `spec.template.metadata.label` in workload which can help trait find the targeted
+pods.
+
+```yaml
+apiVersion: core.oam.dev/v1alpha2
+kind: Component
+metadata:
+  name: example-deployment
+spec:
+  workload:
+    apiVersion: apps/v1
+    kind: Deployment
+    metadata:
+      name: nginx-deployment
+      labels:
+        app: nginx
+    spec:
+      selector:
+        matchLabels:
+          app: nginx
+      template:
+        metadata:
+          labels:
+            component.oam.dev/name: example-deployment
+            app: nginx
+        spec:
+          containers:
+          - name: nginx
+            image: nginx:1.17
+            ports:
+            - containerPort: 80
+```
+
+In all, to achieve various goals, OAM users have to repeatedly specify information to manually establish connections
+between pods, workloads and traits.
+
+## Goals
+
+The goal is to recommend a simple way - Workload Labeling, to easily establish connections between Workloads, Components, AppConfigurations,
+and pods.
+
+## Proposal
+
+Step 1: Setting the label `definition.oam.dev/unique-pod` as the unique identifier of pods
```

Comment:
I suggest we create a new field in workloadDefinition.spec instead of using a label.